File: University/MohasebePichideS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {

            int a = scanner.nextInt();
            int x = scanner.nextInt();
            int n = scanner.nextInt();

            long result = 0;

            for (int k = 0; k <= n; k++) {
                result += (factorial(n))/(factorial(k) * factorial(n-k)) * Math.pow(x, k) * Math.pow(a, n-k);
            }

            System.out.print(result);
        }

    }

    public static long factorial(long n){
        if (n == 0)
            return 1;
        else
            return(n * factorial(n-1));
    }

}

File: University/MohasebePichideS1.py
import math

def main():
    a, x, n = map(int, input().split())
    result = 0
    for k in range(n+1):
        result += (factorial(n) // (factorial(k) * factorial(n-k))) * (x ** k) * (a ** (n-k))
    print(result)

def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)

if __name__ == "__main__":
    main()
